osu.Game.Tests/Visual/Online/TestSceneMessageNotifier.cs

```diff
@@ -0,0 +1,92 @@
+        public void Setup()
+        {
+            friend = new User { Id = 0, Username = "Friend" };
+            publicChannel = new Channel { Id = 1, Name = "osu" };
+            privateMesssageChannel = new Channel(friend) { Id = 2, Name = friend.Username, Type = ChannelType.PM };
+
+            Schedule(() =>
+            {
+                Child = testContainer = new TestContainer(new[] { publicChannel, privateMesssageChannel })
+                {
+                    RelativeSizeAxes = Axes.Both,
+                };
+
+                testContainer.ChatOverlay.Show();
+            });
+        }
+
+        [Test]
+        public void TestPublicChannelMention()
+        {
+            AddStep("Switch to PMs", () => testContainer.ChannelManager.CurrentChannel.Value = privateMesssageChannel);
+
+            AddStep("Send regular message", () => publicChannel.AddNewMessages(new Message(messageIdCounter++) { Content = "Hello everyone!", Sender = friend, ChannelId = publicChannel.Id }));
+            AddAssert("Expect no notifications", () => testContainer.NotificationOverlay.UnreadCount.Value == 0);
+
+            AddStep("Send message containing mention", () => publicChannel.AddNewMessages(new Message(messageIdCounter++) { Content = $"Hello {API.LocalUser.Value.Username.ToLowerInvariant()}!", Sender = friend, ChannelId = publicChannel.Id }));
+            AddAssert("Expect 1 notification", () => testContainer.NotificationOverlay.UnreadCount.Value == 1);
+        }
+
+        [Test]
+        public void TestPrivateMessageNotification()
+        {
+            AddStep("Send PM", () => privateMesssageChannel.AddNewMessages(new Message(messageIdCounter++) { Content = $"Hello {API.LocalUser.Value.Username}!", Sender = friend, ChannelId = privateMesssageChannel.Id }));
+            AddAssert("Expect 1 notification", () => testContainer.NotificationOverlay.UnreadCount.Value == 1);
+        }
+
+        private class TestContainer : Container
+        {
+            private readonly Channel[] channels;
+
+            public TestContainer(Channel[] channels) => this.channels = channels;
+
+            [Cached]
+            public ChannelManager ChannelManager { get; } = new ChannelManager();
+
+            [Cached]
+            public NotificationOverlay NotificationOverlay { get; } = new NotificationOverlay();
+
+            [Cached]
+            public MessageNotifier MessageNotifier { get; } = new MessageNotifier();
+
+            [Cached]
+            public ChatOverlay ChatOverlay { get; } = new ChatOverlay();
+
+            [BackgroundDependencyLoader]
+            private void load()
+            {
+                AddRange(new Drawable[] { ChannelManager, NotificationOverlay });
+
+                ((BindableList<Channel>)ChannelManager.AvailableChannels).AddRange(channels);
+
+                AddRange(new Drawable[] { ChatOverlay, MessageNotifier });
+
+                ((BindableList<Channel>)ChannelManager.JoinedChannels).AddRange(channels);
```

Comment:
I guess you could also call `JoinChannel`. Test code though, so this time it's nothing too big.